plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -315,21 +314,18 @@ public IcebergTableHandle getTableHandle(
                 Optional.empty());
     }
 
-    private long getSnapshotIdFromVersion(Table table, ConnectorTableVersion version)
+    private static long getSnapshotIdFromVersion(Table table, ConnectorTableVersion version)
     {
         io.trino.spi.type.Type versionType = version.getVersionType();
         switch (version.getPointerType()) {
             case TEMPORAL:
-                long epochMillis;
                 if (versionType instanceof TimestampWithTimeZoneType) {
-                    epochMillis = ((TimestampWithTimeZoneType) versionType).isShort()
+                    long epochMillis = ((TimestampWithTimeZoneType) versionType).isShort()
                             ? unpackMillisUtc((long) version.getVersion())
                             : ((LongTimestampWithTimeZone) version.getVersion()).getEpochMillis();
+                    return getSnapshotIdAsOfTime(table, epochMillis);
                 }
-                else {
-                    throw new TrinoException(NOT_SUPPORTED, "Unsupported type for temporal table version: " + versionType.getDisplayName());
-                }
-                return getSnapshotIdAsOfTime(table, epochMillis);
```

Comment:
The code was written like this intentionally, so that the timestamp support can be plugged in later.
We should add timestamp support here instead.